source/Databricks/source/SqlStatementExecution.Tests/ServiceCollectionRegistrationTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿// // Copyright 2020 Energinet DataHub A/S
+// //
+// // Licensed under the Apache License, Version 2.0 (the "License2");
+// // you may not use this file except in compliance with the License.
+// // You may obtain a copy of the License at
+// //
+// //     http://www.apache.org/licenses/LICENSE-2.0
+// //
+// // Unless required by applicable law or agreed to in writing, software
+// // distributed under the License is distributed on an "AS IS" BASIS,
+// // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// // See the License for the specific language governing permissions and
+// // limitations under the License.
+
+using Energinet.DataHub.Core.Databricks.SqlStatementExecution;
+using Energinet.DataHub.Core.Databricks.SqlStatementExecution.Internal.Models;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Energinet.DataHub.Core.Databricks.SqlStatementExecutionTests;
+
+public class ServiceCollectionRegistrationTests
+{
+    [Fact]
+    public void RegisteringDatabricks_Should_ReturnConfiguredHttpClient()
+    {
+        var services = new ServiceCollection();
+        var workspaceUri = "https://foo.com";
+        var workspaceToken = "bar";
+        var warehouseId = "baz";
+
+        services.AddDatabricks<SpySqlStatementClient>(warehouseId, workspaceToken, workspaceUri);
+
+        var serviceProvider = services.BuildServiceProvider();
+
+        var client = serviceProvider.GetRequiredService<ISqlStatementClient>();
+
+        if (client is not SpySqlStatementClient spyClient)
+        {
+            throw new InvalidOperationException("Client is not of type SpySqlStatementClient");
+        }
+
+        Assert.True(spyClient.WorkspaceUriIs(new Uri(workspaceUri)));
+        Assert.True(spyClient.WorkspaceTokenIs(workspaceToken));
```

Comment:
We should be able to use fluent assertions (can be handled in multiple ways, but here is one example):
```suggestion
        using var assertionScope = new AssertionScope();
        var spyClient = client.Should().BeOfType<SpySqlStatementClient>().Subject;
        spyClient.WorkspaceUriIs(new Uri(workspaceUri)).Should().BeTrue();
        spyClient.WorkspaceTokenIs(workspaceToken).Should().BeTrue();
```